pkg/auth/dependency/interaction/provider_commit.go

```diff
@@ -0,0 +1,76 @@
+package interaction
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/skygeario/skygear-server/pkg/auth/dependency/identity"
+	"github.com/skygeario/skygear-server/pkg/auth/model"
+	"github.com/skygeario/skygear-server/pkg/core/auth/metadata"
+	"github.com/skygeario/skygear-server/pkg/core/authn"
+)
+
+func (p *Provider) Commit(i *Interaction) (*authn.Attrs, error) {
+	var err error
+	switch intent := i.Intent.(type) {
+	case *IntentLogin:
+		break
+	case *IntentSignup:
+		err = p.onCommitSignup(i, intent)
+	default:
+		panic(fmt.Sprintf("interaction: unknown intent type %T", i.Intent))
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO(interaction): create new identities & authenticators
+
+	err = p.Store.Delete(i)
+	if err != nil {
+		p.Logger.WithError(err).Warn("failed to cleanup interaction")
+	}
+
+	attrs := &authn.Attrs{
+		UserID:         i.UserID,
+		IdentityType:   i.Identity.Type,
+		IdentityClaims: i.Identity.Claims,
+		// TODO(interaction): populate acr & amr
+	}
+	return attrs, nil
+}
+
+func (p *Provider) onCommitSignup(i *Interaction, intent *IntentSignup) error {
+	if intent.OnUserDuplicate == model.OnUserDuplicateAbort {
```

Comment:
Add a TODO here to remember to handle Merge and CreateNew

Reply:
CreateNew should not need special handling here.
We do not support Merge for signing up, it should be used by OAuth only. It should:
1. Get OAuth data when returning from provider.
2. Create new login interaction.
3. If failed to login (i.e. new user/need merge), create new signup interaction.
4. If failed to sign up (i.e. need merge), create new add identity interaction.

For Auth API, the three interaction flow would be performed automatically.
For Web UI, the final merge (add identity) interaction would be performed only after some kind of confirmation/authentication screen.